src/app/content/practiceQuestions/reducer.spec.ts

```diff
@@ -41,7 +23,40 @@ describe('practice questions reducer', () => {
       ...initialState,
       currentQuestionIndex: 0,
     };
+
     const newState = reducer(state, actions.nextQuestion());
+
     expect(newState.currentQuestionIndex).toEqual(1);
   });
+
+  it('reduces locationChange', () => {
```

Comment:
I'm checking only the properties which should be changed.
```
    expect(newState.currentQuestionIndex).toEqual(initialState.currentQuestionIndex);
    expect(newState.questionAnswers).toEqual(initialState.questionAnswers);
    expect(newState.questions).toEqual(initialState.questions);
    expect(newState.selectedSection).toEqual(initialState.selectedSection);
    expect(newState.loading).toBe(true);
```

Reply:
I see only 
```
    expect(newState.currentQuestionIndex).toEqual(initialState.currentQuestionIndex);
    expect(newState.questionAnswers).toEqual(initialState.questionAnswers);
    expect(newState.questions).toEqual(initialState.questions);
    expect(newState.selectedSection).toEqual(initialState.selectedSection);
    ```
    
and this check will pass because it is not checking for other properties so reducer for `locaitonChange` could update 10 other properties and this check would still pass if properties specified here stayed the same

I think it would be better to check:
```
expect(newState).toEqual({
...initialState,
loading: true,
})
```